components/TextFields/src/ContainedInputView/MDCBaseTextField.m

```diff
@@ -167,22 +145,30 @@ - (MDCBaseTextFieldLayout *)calculateLayoutWithTextFieldSize:(CGSize)textFieldSi
   return [[MDCContainedInputViewVerticalPositioningGuideBase alloc] init];
 }
 
-- (void)layOutLabel {
-  if (self.labelState == MDCContainedInputViewLabelStateFloating) {
-    self.label.font = self.floatingFont;
-    self.label.frame = self.layout.labelFrameFloating;
-  } else {
-    self.label.font = self.normalFont;
-    self.label.frame = self.layout.labelFrameNormal;
-  }
-}
-
 - (CGFloat)clearButtonSideLengthWithTextFieldSize:(CGSize)textFieldSize {
   CGRect bounds = CGRectMake(0, 0, textFieldSize.width, textFieldSize.height);
   CGRect systemPlaceholderRect = [super clearButtonRectForBounds:bounds];
   return systemPlaceholderRect.size.height;
 }
 
+/**
+ To understand this method one must understand that the CGRect UITextField returns from @c
+ -textRectForBounds: does not actually represent the CGRect of visible text in UITextField. It
+ represents the CGRect of an internal "field editing" class, which has a height that is
+ significantly taller than the text (@c font.lineHeight) itself. Providing a height in @c
+ -textRectForBounds: that differs from the height determined by the superclass results in a text
+ field with poor text rendering, sometimes to the point of the text not being visible. By taking the
+ desired CGRect of the visible text from the layout object, giving it the height preferred by the
+ superclass's implementation of @c -textRectForBounds:, and then ensuring that this new CGRect has
+ the same midY as the original CGRect, we are able to take control of the text's positioning.
+ */
+- (CGRect)adjustTextAreaFrame:(CGRect)textRect
+    withParentClassTextAreaFrame:(CGRect)parentClassTextAreaFrame {
+  CGFloat systemDefinedHeight = CGRectGetHeight(parentClassTextAreaFrame);
+  CGFloat minY = CGRectGetMidY(textRect) - (systemDefinedHeight * (CGFloat)0.5);
+  return CGRectMake(CGRectGetMinX(textRect), minY, CGRectGetWidth(textRect), systemDefinedHeight);
+}
+
```

Comment:
Similarly here; this is the second time you've moved this method around :) Please move this out to a separate cleanup PR.